library/network/src/modules/SuSEFirewall.rb

```diff
@@ -2083,6 +2083,86 @@ def GetSupportRoute
       @SETTINGS[:routing]
     end
 
+    def ArePortsOrServicesAllowed(needed_ports, protocol, zone, check_for_aliases)
+      needed_ports = deep_copy(needed_ports)
+      are_allowed = true
+
+      if Ops.less_than(Builtins.size(needed_ports), 1)
+        Builtins.y2warning(
+          "Undefined list of %1 services/ports for service",
+          protocol
+        )
+        return true
+      end
+
+      allowed_ports = {}
+      Ops.set(
+        allowed_ports,
+        "ports",
+        GetAllowedServicesForZoneProto(zone, protocol)
+      )
+
+      Builtins.foreach(needed_ports) do |needed_port|
+        if !Builtins.contains(Ops.get(allowed_ports, "ports", []), needed_port) &&
+           !PortRanges.PortIsInPortranges(
+             needed_port,
+             Ops.get(allowed_ports, "port_ranges", [])
+           )
+          are_allowed = false
+          raise Break
+        end
+      end
+
+      are_allowed
+    end
+
+    # Function returns if requested service is allowed in respective zone.
```

Comment:
This method too is duplicated, where the difference is an omitted block guarded by
```rb
      if !GetProtectFromInternalZone() && Builtins.contains(zones, @int_zone_shortname)
```

If we declare `SuSEFirewalld::GetProtectFromInternalZone` to be always `true` (instead of always false) then we can simply move the original method to the superclass and save 46 lines of code and comments.